src/libs/ReportConnection.ts

```diff
@@ -48,4 +48,24 @@ function getAllReportsLength() {
     return Object.keys(allReports ?? {}).length;
 }
 
-export {getAllReports, getAllReportsNameMap, getAllReportsLength};
+function getReport(reportID: string) {
+    if (!reportID || !allReports) {
+        return;
+    }
+    return allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+}
+
+function updateReportData(reportID: string, reportData?: Partial<Report>) {
```

Comment:
Please remove this function. Any data in Onyx that needs to be updated should be done via one of the proper Onyx methods (merge, set, etc.).

This method will introduce bad side-effects which could have unknown consequences and lead to many bugs without the ability to track them down.

Reply:
FYI I'm working on the [PR](https://github.com/Expensify/App/pull/53385/files/d1399eac1454aa3de78edcdbdaa7c6e15ab45711#diff-b1967016a56ab193c7b36094115b99da05a3edbb8c8c326dc3148320bdb8b593L315-R323) where `ReportConnection` file is removed. There `updateReportData` function usage is replaced with `Onyx.merge`

Reply:
Just to confirm, @stitesExpensify was there any specific reason why you created this function instead of using `Onyx.merge`?